sacred/commandline_options.py

```diff
@@ -8,12 +8,68 @@
 """
 
 import warnings
+from typing import Callable
+import inspect
 
+from sacred.run import Run
 from sacred.commands import print_config
 from sacred.settings import SETTINGS
 from sacred.utils import convert_camel_case_to_snake_case, get_inheritors
 
 
+CLIFunction = Callable[[str, Run], None]
+
+
+class CLIOption:
+    def __init__(
+        self,
+        apply_function: CLIFunction,
+        short_flag: str,
+        long_flag: str,
+        is_flag: bool,
+    ):
+        self.apply_function = apply_function
+        self.short_flag = short_flag
```

Comment:
I think we should check that short flag is a single letter, because docopt will not work correctly otherwise:
`re.match(r'-\w', short_flag)`
Maybe do the same for the long flag too:
`re.match(r'--[\w-]+', long_flag)`

Reply:
Sure things, I'm on holidays, so that might take a while to modify the code.